### ISSUE TITLE
[Bug]: Unable to decode correctly encoded URL segment with a %

### ISSUE THREAD

**Issue by dpyryesk** [remix-run/react-router#11179]
### What version of React Router are you using?

6.21.1

### Steps to Reproduce

1. Use a route with a variable. Example:
```
<Route
    path='customers/:id'
    element={<CustomerDetails/>}
/>
```
2. Navigate to a URL with a correctly encoded ID (`customer-/%`). Example `http://localhost:8080/#/customers/customer-%2F%25`
3. Observe an error
```
The value for the URL param "id" will not be decoded because the string "customer-%2F%" is a malformed URL segment. This is probably due to a bad percent encoding (URIError: URI malformed). 
```

### Expected Behavior

Parameter should be decoded to `customer-/%`

### Actual Behavior

Application throws the following errors:
```
The value for the URL param "*" will not be decoded because the string "customers/customer-%2F%" is a malformed URL segment. This is probably due to a bad percent encoding (URIError: URI malformed). 
    at Routes (http://localhost:8080/static/js/bundle.js:287594:24)
    at Router (http://localhost:8080/static/js/bundle.js:287520:30)
    at HashRouter (http://localhost:8080/static/js/bundle.js:285499:24)
    at Root (http://localhost:8080/static/js/bundle.js:5215:67)
    at ThemeProvider (http://localhost:8080/static/js/bundle.js:113767:24)
    at ApolloProvider (http://localhost:8080/static/js/bundle.js:347696:19)
    at Provider (http://localhost:8080/static/js/bundle.js:282505:20)
    at AppComponent (http://localhost:8080/static/js/bundle.js:7204:21)
    at RootContainer (http://localhost:8080/static/js/bundle.js:27353:66)
```

**Comment by dpyryesk** [comment]
This bug was not present in version 6.4.2. We cannot update to the latest because of this.

**Comment by timdorr** [comment]
Duplicate of #10814

**Comment by timdorr** [comment]
Note there is an outstanding PR to fix this as well: #11138

**Comment by brophdawg11** [comment]
@dpyryesk We have an experimental release out with a potential fix for this - would you want to test your use-case against that version?  Please see details in https://github.com/remix-run/react-router/issues/10814#issuecomment-1896598178 and you can respond back on that open issue as well.  Thanks!
